File: module 6/sprint 2/sprint2-fe/src/app/model/book.ts
import {Author} from './author';
import {Category} from './category';
import {Publisher} from './publisher';

export interface Book {
  id: number;
  bookName: string;
  detail: string;
  size: string;
  totalPage: string;
  quantity: number;
  img: string;
  price: string;
  author: Author;
  category: Category;
  publisher: Publisher;
}

File: module 6/sprint 2/sprint_2_be/src/main/java/com/hongho/model/book/Book.java
package com.hongho.model.book;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "book")
//@JsonIgnoreProperties({"tours"})
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String bookName;
    private String detail;
    private String size;
    private String totalPage;
    private int quantity;
    private String img;
    private String price;

    @ManyToOne
    @JoinColumn(name = "author", referencedColumnName = "id")
    private Author author;

    @ManyToOne
    @JoinColumn(name = "publisher", referencedColumnName = "id")
    private Publisher publisher;

    @ManyToOne
    @JoinColumn(name = "category", referencedColumnName = "id")
    private Category category;
}
